sdk/core/azure-core/src/test/java/com/azure/core/util/logging/ClientLoggerTests.java

```diff
@@ -57,233 +58,167 @@ public void revertLoggingConfiguration() throws Exception {
+     * configured log level.
      */
-    @Test
-    public void logExceptionAsWarningOnlyExceptionMessage() {
+    @ParameterizedTest
+    @MethodSource("logExceptionAsErrorSupplier")
+    public void logExceptionAsError(LogLevel logLevelToConfigure, boolean logContainsMessage,
+        boolean logContainsStackTrace) {
         String exceptionMessage = "An exception message";
-        RuntimeException runtimeException = createRuntimeException(exceptionMessage);
+        IllegalStateException illegalStateException = createIllegalStateException(exceptionMessage);
 
-        String originalLogLevel = setupLogLevel(2);
-        new ClientLogger(ClientLoggerTests.class).logExceptionAsWarning(runtimeException);
-        setPropertyToOriginalOrClear(Configuration.PROPERTY_AZURE_LOG_LEVEL, originalLogLevel);
+        String originalLogLevel = setupLogLevel(logLevelToConfigure.getLogLevel());
+        try {
+            throw new ClientLogger(ClientLoggerTests.class).logExceptionAsError(illegalStateException);
+        } catch (RuntimeException exception) {
+            assertTrue(exception instanceof IllegalStateException);
+        }
+        setPropertyToOriginalOrClear(originalLogLevel);
 
         String logValues = new String(logCaptureStream.toByteArray(), StandardCharsets.UTF_8);
-        assertTrue(logValues.contains(exceptionMessage + System.lineSeparator()));
-        assertFalse(logValues.contains(runtimeException.getStackTrace()[0].toString()));
+        assertEquals(logContainsMessage, logValues.contains(exceptionMessage + System.lineSeparator()));
+        assertEquals(logContainsStackTrace, logValues.contains(illegalStateException.getStackTrace()[0].toString()));
     }
 
     /**
-     * Tests that logging an exception as warning will include the stack trace when the environment log level is set to
-     * VERBOSE.
+     * Tests that logging a Throwable as error will log a message and stack trace appropriately based on the
+     * configured log level.
      */
-    @Test
-    public void logExceptionAsWarningStackTrace() {
+    @ParameterizedTest
+    @MethodSource("logExceptionAsErrorSupplier")
+    public void logCheckedExceptionAsError(LogLevel logLevelToConfigure, boolean logContainsMessage,
+        boolean logContainsStackTrace) {
         String exceptionMessage = "An exception message";
-        RuntimeException runtimeException = createRuntimeException(exceptionMessage);
+        IOException ioException = createIOException(exceptionMessage);
 
-        String originalLogLevel = setupLogLevel(1);
-        new ClientLogger(ClientLoggerTests.class).logExceptionAsWarning(runtimeException);
-        setPropertyToOriginalOrClear(Configuration.PROPERTY_AZURE_LOG_LEVEL, originalLogLevel);
+        String originalLogLevel = setupLogLevel(logLevelToConfigure.getLogLevel());
+        try {
+            throw new ClientLogger(ClientLoggerTests.class).logCheckedExceptionAsError(ioException);
+        } catch (Throwable throwable) {
```

Comment:
Can this be changed to catch IOException instead?

Reply:
I was looking to catch the generic type to ensure that `instanceof` it would be what is expected.